File: App04_News/app/src/main/java/com/callor/topnews/MainActivity.java
package com.callor.topnews;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.callor.topnews.databinding.ActivityMainBinding;
import com.callor.topnews.service.NaverService;
import com.callor.topnews.service.impl.NaverNewsServiceImplV1;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.logging.SimpleFormatter;

public class MainActivity extends AppCompatActivity {
    /**
     * activity.xml 파일에 선언된 view Component를
     * 사용할때 findByViewId를 사용하지 않고
     * 접근할 수 있도록 선언된 경우
     *      buildFeatures{
     *         viewBinding true
     *      }
     * 자동으로 
     * activity_main.xml 파일이름을 확장하여
     * ActivityMainBinding 이라는 클래스가 생성되어 사용할 수 있다
     */
    ActivityMainBinding main_binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        /**
         * 기존에 사용하던
         * setContentView(R.layout.activity_main); 을
         *
         * main_binding = ActivityMainBinding.inflate(getLayoutInflater());
         * setContentView(main_binding.getRoot());
         * 코드로 변경
         *
         * 이 코드로 시작이 되면 activity.xml 파일에 있는
         * 모든 view Component가 한꺼번에 사용가능한 상태로 변경된다
         * (findByViewid를 사용하지 않아도 된다)
         */
        main_binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(main_binding.getRoot());

        Date date = new Date(System.currentTimeMillis());
        SimpleDateFormat sd = new SimpleDateFormat("yyyy-MM-dd");
        String curDate = sd.format(date);

        NaverService naverService = new NaverNewsServiceImplV1(main_binding.newsListView);
        naverService.getNews(curDate);

    }
}

File: HelloApp3/app/src/main/java/com/callor/hello/MainActivity.kt
package com.callor.hello

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.callor.hello.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    // lateinit var : 지금변수를 선언만하고
    // 생성은 잠시후에 실행하겠다
    private lateinit var mainBiding:ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 선언된 mainBinding 을 activity_main.xml 파일을 열어서
        // inflate(확장) 하여 mainBinding 객체로(변수로) 초기화
        mainBiding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(mainBiding.root)

        val navController = findNavController(R.id.fragmentContainer)
        val appBarConfig = AppBarConfiguration(
            setOf(
                R.id.bottom_nav_home,R.id.bottom_nav_book
            )
        )
        setupActionBarWithNavController(navController,appBarConfig)
        mainBiding.bottomNav.setupWithNavController(navController)

    }
}